### PR TITLE
Iss#87 house member documents integration tests

### LINKED ISSUE #87: House member documents integration tests
## ⚠️ Is your feature request related to a problem? Please describe

add integration tests for `HouseMemberDocumentController` and `HouseMemberDocumentService`

## 🤚 Do you want to develop this feature yourself?

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by DDmit04**
## 🚀 Description

Add integration tests for `HouseMemberDocumentController` and `HouseMemberDocumentService`

## 📄 Motivation and Context

Closes #87 

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/com/myhome/bootstrap/DataLoader.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
 
   @Override public void run(String... args) throws Exception {
     //loadPostmanData();
-    loadData();
+    //loadData();
```
Comment (DDmit04) [inline src/main/java/com/myhome/bootstrap/DataLoader.java]: Strange thing here - if uncomment lines and run all test in IDEA, then lots of them fails, but all pull request checks are passed

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: Wildcard imports.
https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

Please setup IntelliJ correctly
https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+@TestPropertySource(properties = {
+    "files.maxSizeKBytes=1",
+    "files.compressionBorderSizeKBytes=99",
+    "files.compressedImageQuality=0.99"
+})
+public class HouseMemberDocumentServiceIntegrationTest extends ControllerIntegrationTestBase {
+
+  private static final String MEMBER_ID = "default-member-id-for-testing";
+  private static final String TEST_DOCUMENT_NAME = "test-document";
+  private static final String testUserName = "Test User";
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: `static final` fields should be upper case like `MEMBER_ID` and `TEST_DOCUMENT_NAME`

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+  @Autowired
+  private HouseMemberRepository houseMemberRepository;
+  @Autowired
+  private UserRepository userRepository;
+  @Autowired
+  private MockMvc mockMvc;
+
+  @BeforeEach
+  void setUp() {
+    if(!houseMemberRepository.findByMemberId(MEMBER_ID).isPresent()) {
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: Use @DirtiesContext instead please
https://www.baeldung.com/spring-dirtiescontext

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    HouseMember member = houseMemberRepository.findByMemberId(MEMBER_ID).get();
+    member.setHouseMemberDocument(null);
+    houseMemberRepository.save(member);
+  }
+
+  @Test
+  void getHouseMemberDocumentSuccess() throws Exception {
+    // given
+    addDefaultHouseMemberDocument();
+    // when
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: Please leave one blank line between given-when and when-then.
Applicable for all occurences

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    MvcResult mvcResult = mockMvc.perform(get("/members/{memberId}/documents", MEMBER_ID)
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: Let's extract this url `/members/{memberId}/documents` to `private static const` variable

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        .headers(getHttpEntityHeaders()))
+        .andDo(print())
+        .andExpect(status().isOk())
+        .andReturn();
+
+    // then
+    HouseMember member = houseMemberRepository.findByMemberId(MEMBER_ID).get();
+    assertEquals(mvcResult.getResponse().getContentType(), MediaType.IMAGE_JPEG_VALUE);
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: `mvcResult.getResponse()` extract to variable

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+    HouseMember member = houseMemberRepository.findByMemberId(MEMBER_ID).get();
+    assertNull(member.getHouseMemberDocument());
+  }
+
+  @Test
+  void deleteHouseMemberDocumentMemberNotExists() throws Exception {
+    // given
+    addDefaultHouseMemberDocument();
+
+    mockMvc.perform(delete("/members/{memberId}/documents", "non-existing-member-id")
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: `// when`

**Review by mslowiak — CHANGES_REQUESTED** [review]

File: src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java (hunk 2 above)
Comment (DDmit04) [reply src/test/java/com/myhome/services/integration/HouseMemberDocumentServiceIntegrationTest.java]: Thanks

**Comment by DDmit04** [comment]
with `@DirtiesContext` annotation added tests takes more time to execute

**Comment by DDmit04** [comment]
With `@DirtyesContext` use build takes 65 min

**Comment by DDmit04** [comment]
without `@DirtyesContext` - 10 min

**Comment by DDmit04** [comment]
I created 12 test cases. I need refrash application context after each (manually or with `@DirtyesContext`). With `@DirtyesContext` spring 12 times restarts all app that why tests runs too long. Can I leave this code segment without using `@DirtyesContext`?
```
@BeforeEach
  void setUp() {
    if(!houseMemberRepository.findByMemberId(MEMBER_ID).isPresent()) {
      HouseMember member = houseMemberRepository.save(new HouseMember(MEMBER_ID, null, "test-member-name", null));
      member.setMemberId(MEMBER_ID);
      houseMemberRepository.save(member);
    }
    authDefaultUser();
  }

  @AfterEach()
  void cleanHouseMemberDocument() {
    HouseMember member = houseMemberRepository.findByMemberId(MEMBER_ID).get();
    member.setHouseMemberDocument(null);
    houseMemberRepository.save(member);
  }
```

**Comment by jmprathab** [comment]
> I created 12 test cases. I need refrash application context after each (manually or with `@DirtyesContext`). With `@DirtyesContext` spring 12 times restarts all app that why tests runs too long. Can I leave this code segment without using `@DirtyesContext`?
> 
> ```
> @BeforeEach
>   void setUp() {
>     if(!houseMemberRepository.findByMemberId(MEMBER_ID).isPresent()) {
>       HouseMember member = houseMemberRepository.save(new HouseMember(MEMBER_ID, null, "test-member-name", null));
>       member.setMemberId(MEMBER_ID);
>       houseMemberRepository.save(member);
>     }
>     authDefaultUser();
>   }
> 
>   @AfterEach()
>   void cleanHouseMemberDocument() {
>     HouseMember member = houseMemberRepository.findByMemberId(MEMBER_ID).get();
>     member.setHouseMemberDocument(null);
>     houseMemberRepository.save(member);
>   }
> ```

Without `@DirtiesContext` should be fine.

**Comment by DDmit04** [comment]
It is done i think

**Review by mslowiak — APPROVED** [review]
Great job!

**Pull request merged**